modules/data.land/R/download.SM_CDS.R

```diff
@@ -0,0 +1,91 @@
+#' Download CDS soil moisture data for the SDA workflow.
+#'
+#' @param outfolder physical paths to where the unziped soil moisture files are downloaded.
+#' @param time_points A vector contains each time point within the start and end date.
+#' @param overwrite flag determine if we want to overwrite existing files when downloading.
+#'
+#' @return A vector containing file paths to the downloaded files.
+#' @export
+#' 
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom magrittr %>%
+download.SM_CDS <- function(outfolder, time.points, overwrite = FALSE) {
+  #load cdsapi from python environment.
+  tryCatch({
+    cdsapi <- reticulate::import("cdsapi")
+  }, error = function(e) {
+    PEcAn.logger::logger.severe(
+      "Failed to load `cdsapi` Python library. ",
+      "Please make sure it is installed to a location accessible to `reticulate`.",
+      "You should be able to install it with the following command: ",
+      "`pip install --user cdsapi`.",
+      "The following error was thrown by `reticulate::import(\"cdsapi\")`: ",
+      conditionMessage(e)
+    )
+  })
+  #check if the token exists for the cdsapi.
+  if (!file.exists(file.path(Sys.getenv("HOME"), ".cdsapirc")))
```

Comment:
Likewise, you definitely need to document the need to set up an api key as part of the function documentation. All these error messages are great, but it would be super frustrating to try to use this function in practice as it would just keep giving you different error messages until you finally got it working. I suspect most users would give up before they got through all the dependencies.